cpu/pic32/lib/pic32_i2c.c

```diff
@@ -118,7 +118,13 @@
     IFS1CLR = (_IFS1_I2C##XX##MIF_MASK ) | (_IFS1_I2C##XX##BIF_MASK );   \
     I2C##XX##CON = 0;                                                    \
     I2C##XX##CONSET = _I2C##XX##CON_SMEN_MASK;                           \
-                                                                         \
+    return 0;                                                            \
+  }                                                                      \
+  uint8_t                                                                \
+  pic32_i2c##XX##_release(void)                                          \
+  {                                                                      \
+    pic32_i2c##XX##_power_down();                                        \
```

Comment:
Power down needs to be done after register changes, otherwise any register changes will be discarded.